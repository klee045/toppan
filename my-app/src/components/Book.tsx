import React, { CSSProperties } from "react";
import { RiArrowDropDownLine, RiArrowDropUpLine } from "react-icons/ri";
import BorrowerList from "./BorrowerList";

const BookStyle: CSSProperties = {
  padding: "30px",
  borderRadius: "10px",
  boxShadow: "3px 6px 5px gray",
  backgroundColor: "rgb(241, 230, 230)",
  width: "100%",
};

const fontSize: string = "40px";

const BookDescriptionStyle: CSSProperties = {
  display: "flex",
  flexDirection: "row",
  justifyContent: "space-between",
};

const IdxDivStyle: CSSProperties = {
  fontSize: fontSize,
  marginLeft: "28px",
};

const BookNameDivStyle: CSSProperties = {
  fontSize: fontSize,
};

const ArrowDivStyle: CSSProperties = {
  fontSize: fontSize,
  marginRight: "28px",
  display: "flex",
  alignItems: "center",
};

const AuthorDivStyle: CSSProperties = {
  marginRight: "28px",
  textAlign: "end",
};

const Book = ({
  idx,
  bookName,
  authorName,
  borrowers,
  isToggled,
}: {
  idx: number;
  bookName: string;
  authorName: string;
  borrowers: string[];
  isToggled: boolean;
}): JSX.Element => {
  return (
    <>
      <div style={BookStyle}>
        <div style={BookDescriptionStyle}>
          <div style={IdxDivStyle}>{idx}</div>
          <div style={BookNameDivStyle}>{bookName}</div>
          <div style={ArrowDivStyle}>
            {isToggled ? <RiArrowDropUpLine /> : <RiArrowDropDownLine />}
          </div>
        </div>
        <div style={AuthorDivStyle}>by {authorName}</div>
      </div>

      {isToggled ? <BorrowerList borrowers={borrowers} /> : <></>}
    </>
  );
};

export default Book;
