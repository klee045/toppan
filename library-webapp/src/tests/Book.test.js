import React from "react";
import renderer from "react-test-renderer";
import Book from "../components/Book";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Book
        idx={1}
        bookName={"The Stone Sky"}
        authorName={"N.K. Jemisin"}
        borrowers={[]}
        isToggled={false}
        handleBookClick={""}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
