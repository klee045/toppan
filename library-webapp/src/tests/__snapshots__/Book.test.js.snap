// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  onClick={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(241, 230, 230)",
      "borderRadius": "10px",
      "boxShadow": "3px 6px 5px gray",
      "padding": "30px",
      "width": "100%",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <div
      style={
        Object {
          "fontSize": "40px",
          "marginLeft": "28px",
        }
      }
    >
      1
    </div>
    <div
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      The Stone Sky
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "fontSize": "40px",
          "marginRight": "28px",
        }
      }
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 15l-4.243-4.243 1.415-1.414L12 12.172l2.828-2.829 1.415 1.414z"
          />
        </g>
      </svg>
    </div>
  </div>
  <div
    style={
      Object {
        "marginRight": "28px",
        "textAlign": "end",
      }
    }
  >
    by 
    N.K. Jemisin
  </div>
</div>
`;
